### PR TITLE
Add perl argument to matches()

### LINKED ISSUE #71: Feature request: add argument `perl` in `matches`
Add possibility to use Perl-like regular expressions in `matches()`

``` r
iris$S <- 10
dplyr::select(iris[1,], tidyselect::matches("^s(?!e)"))
#> Error in grep(needle, haystack, ...): invalid regular expression '^s(?!e)', reason 'Invalid regexp'
```

I'm asking to add argument `perl` only and not `...` as in #58

### PR DESCRIPTION
**Pull request by fmichonneau**
adds the option to use Perl regular expressions to select variables.

Fixes #71

### CODE REVIEW

**Comment by hadley** [comment]
(tweaked your title because `Fixes #71` needs to go in the description, not the title)

File: NEWS.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 
 * `vars_select()` ignores vectors with only zeros (#82). 
 
+* `matches()` has new argument `perl` to allow for Perl-like regular expressions (requested by @GegznaV, #71)
```
Comment (hadley) [inline NEWS.md]: Can you please follow the style in https://style.tidyverse.org/news.html#bullets? (i.e. it should mention you as the PR writer, not the original issue creator)

Comment (fmichonneau) [reply NEWS.md]: thanks, done!

**Comment by hadley** [comment]
Thanks @fmichonneau

**Pull request merged**